csp-mixin/signals/virtualMachines.libsonnet

```diff
@@ -0,0 +1,141 @@
+local commonlib = import 'common-lib/common/main.libsonnet';
+function(this)
+  {
+    local s = self,
+    filteringSelector: this.filteringSelector,
+    groupLabels: this.groupLabels,
+    instanceLabels: this.instanceLabels,
+    aggLevel: 'instance',
+    discoveryMetric: {
+      azuremonitor: 'azure_microsoft_compute_virtualmachines_vmavailabilitymetric_average_count',
+    },
+    signals: {
+      cpuUtilization: {
+        name: 'CPU Utilization average',
+        description: 'The percentage of allocated compute units that are currently in use by the Virtual Machine(s)',
+        type: 'raw',
+        unit: 'percent',
+        sources: {
+          azuremonitor: {
+            expr: 'avg by (resourceName, job, resourceGroup, subscriptionName) (azure_microsoft_compute_virtualmachines_percentage_cpu_average_percent{%(queriesSelector)s})',
+            legendCustomTemplate: '{{resourceName}}',
+          },
+        },
+      },
+
+      availableMemory: {
+        name: 'Available memory',
+        description: 'Amount of physical memory, in bytes, immediately available for allocation to a process or for system use in the Virtual Machine',
+        type: 'raw',
+        unit: 'decbytes',
+        sources: {
+          azuremonitor: {
+            expr: 'sum by (resourceName, job, resourceGroup, subscriptionName) (azure_microsoft_compute_virtualmachines_available_memory_bytes_average_bytes{%(queriesSelector)s})',
+            legendCustomTemplate: '{{resourceName}}',
+          },
+        },
+      },
+      cpuCreditsConsumed: {
+        name: 'CPU Credits Consumed',
+        description: 'Total number of credits consumed by the Virtual Machine.',
+        type: 'raw',
+        unit: 'short',
+        sources: {
+          azuremonitor: {
+            expr: 'sum by (resourceName, job, resourceGroup, subscriptionName) (azure_microsoft_compute_virtualmachines_cpu_credits_consumed_average_count{%(queriesSelector)s})',
+            legendCustomTemplate: '{{resourceName}}',
+          },
+        },
+      },
+      cpuCreditsRemaining: {
+        name: 'CPU Credits Remaining',
```

Comment:
Best if we keep our titles consistent. Here is the Title Case, while next signal has Sentence case name. According to grafana writers guide, Sentence case should be used everywhere.